Sources/_StringProcessing/Engine/Structuralize.swift

```diff
@@ -1,20 +1,38 @@
 internal import _RegexParser
 
-extension CaptureList {
-  @available(SwiftStdlib 5.7, *)
-  func createElements(
-    _ list: MECaptureList
+@available(SwiftStdlib 5.7, *)
+extension Executor {
+  static func createExistentialElements(
+    _ program: MEProgram,
+    matchRange: Range<String.Index>,
+    storedCaptures: [Processor._StoredCapture],
+    wholeMatchValue: Any?
   ) -> [AnyRegexOutput.ElementRepresentation] {
-    assert(list.values.count == captures.count)
-    
+    let capList = program.captureList
+    let capOffsets = program.referencedCaptureOffsets
+
+    // Formal captures include the entire match
+    assert(storedCaptures.count + 1 == capList.captures.count)
+
     var result = [AnyRegexOutput.ElementRepresentation]()
-    
-    for (i, (cap, meStored)) in zip(captures, list.values).enumerated() {
+    result.append(
```

Comment:
We should be able to reserve the correct capacity in `result` before this…

```suggestion
    result.reserveCapacity(capList.captures.count)
    result.append(
```